src/core/server/saved_objects/migrationsv2/migrations_state_action_machine.ts

```diff
@@ -35,9 +41,15 @@ const logStateTransition = (
   newState: State
 ) => {
   if (newState.logs.length > oldState.logs.length) {
-    newState.logs
-      .slice(oldState.logs.length)
-      .forEach((log) => logger[log.level](logMessagePrefix + log.message));
```

Comment:
Spent awhile battling TS on this (and a similar situation in the `actions` plugin).

The issue is that dynamically calling a method from `Logger` results in an `This expression is not callable` error due to each member of the union type having signatures which are incompatible (similar to [this issue](https://github.com/microsoft/TypeScript/issues/33591)).

Specifically, it seems that when you are mixing methods which could contain `string | Error` in the message (warn, error, fatal) with ones that only contain `string` (trace, debug, info), TS freaks out.

But interestingly, this only surfaced when I added the generic type params to the `Logger` interface... removing them resolves the issue.

Still not sure what's up with that, but would appreciate any insights folks may have. In the meantime, I resorted to casting in both places 😞